src/app/services/test-records/test-records.service.ts

```diff
@@ -134,4 +137,25 @@ export class TestRecordsService {
   get isSameTestTypeId$(): Observable<boolean> {
     return this.store.pipe(select(isTestTypeKeySame('testTypeId')));
   }
+
+  testTypeChange(testTypeId: string) {
+    this.store.dispatch(testTypeIdChanged({ testTypeId }));
+  }
+
+  get isTestTypeGroupEditable$() {
+    return this.store.pipe(select(toEditOrNotToEdit)).pipe(
```

Comment:
Could put all this logic inside a selector?

Reply:
It could although I see no benefit. I prefer to let the selector fetch state and service can orchestrate logic.

Reply:
fair enough